index.js

```diff
@@ -32,10 +37,30 @@ const slowDownPlugin = async (fastify, settings) => {
       remaining: remainingRequests
     }
 
+    if (options.skipFailedRequests) {
+      onFinished(reply.raw, err => {
+        if (err || reply.statusCode >= 400) {
+          store.decrementOnKey(key)
+        }
+      })
+    }
+
+    if (options.skipSuccessfulRequests) {
+      onFinished(reply.raw, err => {
+        if (!err && reply.statusCode < 400) {
+          store.decrementOnKey(key)
+        }
+      })
+    }
+
     if (!hasDelay) {
       return
     }
 
+    if (requestCounter - 1 === options.delayAfter) {
+      options.onLimitReached(req, reply, options)
```

Comment:
IMO, as this is an additional feature, you should not consider `onLimitReached` a mandatory field.
Otherwise, this will result in a breaking change and a major version must be released.

Reply:
They are being set as no-op functions as default: https://github.com/nearform/fastify-slow-down/pull/23/files#diff-602f56bb692d9bf3bdc5fef3d75d5fa33fe841c3d21ca46294bb6fdaf0785d0cR10-R13